src/disasm.cpp

```diff
@@ -65,14 +65,367 @@
+        Out << LineStart << " Function " << frame.FunctionName;
+        if (bracket_outer || i != 0)
+            Out << " {";
+        Out << "\n" << LineStart << " Location: " << frame.FileName;
+        if (frame.Line != UINT_MAX && frame.Line != 0)
+            Out << ":" << frame.Line;
+        Out << "\n";
+    }
+}
+
+
+// adaptor class for printing line numbers before llvm IR lines
+class LineNumberAnnotatedWriter : public AssemblyAnnotationWriter {
+    DILocation *InstrLoc = nullptr;
+    DILineInfoPrinter LinePrinter{';', false};
+    DenseMap<const Instruction *, DILocation *> DebugLoc;
+    DenseMap<const Function *, DISubprogram *> Subprogram;
+public:
+    LineNumberAnnotatedWriter() {}
+    virtual void emitFunctionAnnot(const Function *, formatted_raw_ostream &);
+    virtual void emitInstructionAnnot(const Instruction *, formatted_raw_ostream &);
+    virtual void emitBasicBlockEndAnnot(const BasicBlock *, formatted_raw_ostream &);
+    // virtual void printInfoComment(const Value &, formatted_raw_ostream &) {}
+
+    void addSubprogram(const Function *F, DISubprogram *SP)
+    {
+        Subprogram[F] = SP;
+    }
+
+    void addDebugLoc(const Instruction *I, DILocation *Loc)
+    {
+        DebugLoc[I] = Loc;
+    }
+};
+
+void LineNumberAnnotatedWriter::emitFunctionAnnot(
+      const Function *F, formatted_raw_ostream &Out)
+{
+    InstrLoc = nullptr;
+    DISubprogram *FuncLoc = F->getSubprogram();
+    if (!FuncLoc) {
+        auto SP = Subprogram.find(F);
+        if (SP != Subprogram.end())
+            FuncLoc = SP->second;
+    }
+    if (!FuncLoc)
+        return;
+    std::vector<DILineInfo> DIvec(1);
+    DILineInfo &DI = DIvec.back();
+    DI.FunctionName = FuncLoc->getName();
+    DI.FileName = FuncLoc->getFilename();
+    DI.Line = FuncLoc->getLine();
+    LinePrinter.emit_lineinfo(Out, DIvec);
+}
+
+void LineNumberAnnotatedWriter::emitInstructionAnnot(
+      const Instruction *I, formatted_raw_ostream &Out)
+{
+    DILocation *NewInstrLoc = I->getDebugLoc();
+    if (!NewInstrLoc) {
+        auto Loc = DebugLoc.find(I);
+        if (Loc != DebugLoc.end())
+            NewInstrLoc = Loc->second;
+    }
+    if (!NewInstrLoc || NewInstrLoc == InstrLoc)
+        return;
+    InstrLoc = NewInstrLoc;
+    std::vector<DILineInfo> DIvec;
+    do {
+        DIvec.emplace_back();
+        DILineInfo &DI = DIvec.back();
+        DIScope *scope = NewInstrLoc->getScope();
+        if (scope)
+            DI.FunctionName = scope->getName();
+        DI.FileName = NewInstrLoc->getFilename();
+        DI.Line = NewInstrLoc->getLine();
+        NewInstrLoc = NewInstrLoc->getInlinedAt();
+    } while (NewInstrLoc);
+    LinePrinter.emit_lineinfo(Out, DIvec);
+}
+
+void LineNumberAnnotatedWriter::emitBasicBlockEndAnnot(
+        const BasicBlock *BB, formatted_raw_ostream &Out)
+{
+    if (BB == &BB->getParent()->back())
+        LinePrinter.emit_finish(Out);
+}
+
+// print an llvm IR acquired from jl_get_llvmf
```

Comment:
I think `print` (as in `-to-string`) is sufficiently clear as a verb, in context here.